src/Data/Singletons/Internal.hs

```diff
@@ -178,6 +222,72 @@ infixl 9 @@
 newtype instance Sing (f :: k1 ~> k2) =
   SLambda { applySing :: forall t. Sing t -> Sing (f @@ t) }
 
+-- | An (explicitly bidirectional) pattern synonym for
```

Comment:
Also, these Haddocks are only attached to `SLambda2`, despite the fact that it equally applies to every `SLambda*` pattern synonym. In light of this, I would suggest turning this into a Haddock block, like so:

```haskell
-- (In the export list)
-- $SLambdaPatternSynonyms
  SLambda2, SLambda3, ...

{- $SLambdaPatternSynonyms

@SLambda{2...8}@ are explicitly bidirectional pattern synonyms for...
-}
```